src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -311,7 +311,8 @@ export async function makeUtxoEngine(
             )
             targets.push({
               address: publicAddress,
-              value: parseInt(target.nativeAmount)
+              value: parseInt(target.nativeAmount),
+              memo: target.memo == null ? undefined : target.memo
```

Comment:
No need to check if it's null here, just add it directly.
You are testing for null anyway in the createTx function anyway.
About the type issue, see my next comment.